nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/test/java/org/apache/nifi/record/script/ScriptedRecordSetWriterTest.java

```diff
@@ -0,0 +1,130 @@
+        final TestRunner runner = TestRunners.newTestRunner(new AbstractProcessor() {
+            @Override
+            public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+            }
+        });
+
+        MockScriptedWriter recordSetWriterFactory = new MockScriptedWriter();
+        runner.addControllerService("writer", recordSetWriterFactory);
+        runner.setProperty(recordSetWriterFactory, "Script Engine", "Groovy");
+        runner.setProperty(recordSetWriterFactory, ScriptingComponentUtils.SCRIPT_FILE, targetPath.toString());
+        runner.setProperty(recordSetWriterFactory, ScriptingComponentUtils.SCRIPT_BODY, (String) null);
+        runner.setProperty(recordSetWriterFactory, ScriptingComponentUtils.MODULES, (String) null);
+        runner.enableControllerService(recordSetWriterFactory);
+
+        RecordSchema schema = recordSetWriterFactory.getSchema(Collections.emptyMap(), null);
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        RecordSetWriter recordSetWriter = recordSetWriterFactory.createWriter(new MockComponentLog("id", recordSetWriterFactory), schema, outputStream, Collections.emptyMap());
+        assertNotNull(recordSetWriter);
+
+        SimpleRecordSchema recordSchema = new SimpleRecordSchema(Arrays.asList(new RecordField("id", RecordFieldType.INT.getDataType()),
+                new RecordField("name", RecordFieldType.STRING.getDataType()),
+                new RecordField("code", RecordFieldType.INT.getDataType())));
+        MapRecord [] records = createMapRecords(recordSchema);
+
+        recordSetWriter.write(RecordSet.of(recordSchema, records));
+
+        DocumentBuilder documentBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+        Document document = documentBuilder.parse(new ByteArrayInputStream(outputStream.toByteArray()));
+        XPathFactory xpathfactory = XPathFactory.newInstance();
+        XPath xpath = xpathfactory.newXPath();
+        assertEquals("1", xpath.evaluate("//record[1]/id/text()", document));
+        assertEquals("200", xpath.evaluate("//record[2]/code/text()", document));
+        assertEquals("Ramon", xpath.evaluate("//record[3]/name/text()", document));
+    }
+
+    private static MapRecord[] createMapRecords(SimpleRecordSchema recordSchema) {
+        Map<String, Object> map = new LinkedHashMap<>(3);
+        map.put("id", 1);
+        map.put("name", "John");
+        map.put("code", 100);
+        Map<String, Object> map1 = new LinkedHashMap<>(3);
+        map1.put("id", 2);
+        map1.put("name", "Mary");
+        map1.put("code", 200);
+        Map<String, Object> map2 = new LinkedHashMap<>(3);
+        map2.put("id", 3);
+        map2.put("name", "Ramon");
+        map2.put("code", 300);
+
+        return  new MapRecord[]{new MapRecord(recordSchema, map), new MapRecord(recordSchema, map1), new MapRecord(recordSchema, map2)};
```

Comment:
```suggestion
        return new MapRecord[]{new MapRecord(recordSchema, map), new MapRecord(recordSchema, map1), new MapRecord(recordSchema, map2)};
```

Reply:
@exceptionfactory I am sorry can you please explain what the suggested change here is? This what I am seeing which looks identical.
![image](https://github.com/apache/nifi/assets/8440116/737680ef-c443-4301-862c-839ab5217d24)

Reply:
Sorry for the lack of clarity, there was just an extra space between `return` and `new`.